pkg/skaffold/build/scheduler_test.go

```diff
@@ -341,17 +376,29 @@ func TestInParallelForArgs(t *testing.T) {
 				artifacts[i] = &latest.Artifact{ImageName: a}
 				tags[a] = fmt.Sprintf("%s@tag%d", a, i+1)
 			}
-			if test.inSeqFunc != nil {
-				t.Override(&runInSequence, test.inSeqFunc)
-			}
+
+			setDependencies(artifacts, test.dependency)
 			initializeEvents()
-			actual, _ := InParallel(context.Background(), ioutil.Discard, tags, artifacts, test.buildArtifact, 0)
+			actual, _ := InOrder(context.Background(), ioutil.Discard, tags, artifacts, test.buildArtifact, test.concurrency)
 
 			t.CheckDeepEqual(test.expected, actual)
 		})
 	}
 }
 
+func setDependencies(a []*latest.Artifact, d [][]int) {
+	for _, dep := range d {
+		for i := range dep {
+			if i == 0 {
+				continue
+			}
+			a[dep[0]-1].Dependencies = append(a[dep[0]-1].Dependencies, &latest.ArtifactDependency{
```

Comment:
just by pattern matching with the next line, should this be `a[dep[i]-1]`?

Reply:
improved the test